test/operator/test_fermions.py

```diff
@@ -893,26 +931,14 @@ def test_fermion_reduce():
         constant=1,
     )
     op1_ordered = op1.to_normal_order()
-    op2 = nkx.operator.FermionOperator2nd(
-        hi, terms=("2^ 1^ 0^ 2 1 0", "1^ 0^ 1 0"), weights=(-1, 1), constant=1
-    )
+    op2 = nkx.operator.FermionOperator2nd(hi, terms=[""], weights=[1])
```

Comment:
Was this a bug? Before the test was comparing 
```python
 op1 = nkx.operator.FermionOperator2nd(
        hi,
        terms=("0 0^ 0 0^ 1^ 1 0^ 0 2 2^ 2 2^ 2 2^",),
        constant=1,
    )
# which corresponds to
>>> In [5]: print(op1.to_normal_order().operator_string())
1.0 []
In [6]: op1.to_dense()
Out[6]:
array([[1., 0., 0., 0., 0., 0., 0., 0.],
       [0., 1., 0., 0., 0., 0., 0., 0.],
       [0., 0., 1., 0., 0., 0., 0., 0.],
       [0., 0., 0., 1., 0., 0., 0., 0.],
       [0., 0., 0., 0., 1., 0., 0., 0.],
       [0., 0., 0., 0., 0., 1., 0., 0.],
       [0., 0., 0., 0., 0., 0., 1., 0.],
       [0., 0., 0., 0., 0., 0., 0., 1.]])
```
with
```python
    op2 = nkx.operator.FermionOperator2nd(
        hi, terms=("2^ 1^ 0^ 2 1 0", "1^ 0^ 1 0"), weights=(-1, 1), constant=1
    )
# which corresponds to
In [7]: print(op2.to_normal_order().operator_string())
1 [] +
-1 [2^ 1^ 0^ 2 1 0] +
1 [1^ 0^ 1 0]
In [8]: op2.to_dense()
Out[8]:
array([[1, 0, 0, 0, 0, 0, 0, 0],
       [0, 1, 0, 0, 0, 0, 0, 0],
       [0, 0, 1, 0, 0, 0, 0, 0],
       [0, 0, 0, 1, 0, 0, 0, 0],
       [0, 0, 0, 0, 1, 0, 0, 0],
       [0, 0, 0, 0, 0, 1, 0, 0],
       [0, 0, 0, 0, 0, 0, 0, 0],
       [0, 0, 0, 0, 0, 0, 0, 1]])
```

while now you correctly compare it to 
```python
 op2 = nkx.operator.FermionOperator2nd(hi, terms=[""], weights=[1])
```
which is the identity....

So this PR is also fixing a bug, which we were improperly testing...?

Can you confirm?

Reply:
Yes, indeed, on current master( not this PR) I get

```python
In [8]:  op1 = nkx.operator.FermionOperator2nd(
   ...:         hi,
   ...:         terms=("0 0^ 0 0^ 1^ 1 0^ 0 2 2^ 2 2^ 2 2^",),
   ...:         constant=1,
   ...:     )
   ...:

In [9]: print(op1.to_normal_order().operator_string())
1.0 [] +
1.0 [1^ 0^ 1 0] +
-1.0 [2^ 1^ 0^ 2 1 0]

In [10]: op1.to_dense()
    ...:
Out[10]:
array([[1., 0., 0., 0., 0., 0., 0., 0.],
       [0., 1., 0., 0., 0., 0., 0., 0.],
       [0., 0., 1., 0., 0., 0., 0., 0.],
       [0., 0., 0., 1., 0., 0., 0., 0.],
       [0., 0., 0., 0., 1., 0., 0., 0.],
       [0., 0., 0., 0., 0., 1., 0., 0.],
       [0., 0., 0., 0., 0., 0., 1., 0.],
       [0., 0., 0., 0., 0., 0., 0., 1.]])
```

so the connected elements must be correct, but the normal ordering seems to be wrong..

Reply:
Yes, there was a bug in here before, so definitely keep the new tests.